### PR TITLE
Stoppable server

### PR DESCRIPTION
**Pull request by rushsteve1**
## Summary
Added the `Server::stoppable()` function which provides a method for gracefully stopping a `Server`.

## Motivation
There was obvious way to gracefully stop a Rouille server, and I ended up implementing this code in my own project. Since it's fairly simple and a useful feature I figured I would contribute it back.

See the provided documentation example for basic usage.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -361,6 +362,46 @@ impl<F> Server<F> where F: Send + Sync + 'static + Fn(&Request) -> Response {
...
+    /// handle.join().unwrap();
+    /// ```
+    #[inline]
+    pub fn stoppable(self) -> (thread::JoinHandle<()>, mpsc::Sender<()>) {
+        let (tx, rx) = mpsc::channel();
+
+        let handle = thread::spawn(move || {
+            while let Err(_) = rx.try_recv() {
+                self.poll();
+            }
```
Comment (LPGhatguy) [inline src/lib.rs]: Won't this thread eat your CPU relentlessly?

Comment (rushsteve1) [reply src/lib.rs]: Yes actually, thank you for pointing that out.

**Comment by rushsteve1** [comment]
As @LPGhatguy pointed out it was consuming a lot of CPU. I fixed that with c1256a490d37a55550f30a3fe8ce9f3c9454a869

As it stands on my machine the system monitor reports 0% CPU usage when idle and ~50% when under heavy synthetic load.

The only caveat is that there is at most 1 second between checks to see if the stop has been requested. There is no delay between server responses however. I believe that 1s can be lowered but I thought it was a good conservative delay.

[Linked here is a simple project with a synthetic benchmark for testing purposes.](https://rushsteve1.us/upload/f.php?h=2GvOnxfX)

**Comment by rushsteve1** [comment]
@tomaka Are there any further issues with this? If not I would like to get it merged.

**Comment by rushsteve1** [comment]
I just realized that this PR is a (better) alternative to #203 

Also @tomaka again can I please get a response or something? This has been just sitting for a few weeks now.

**Comment by rushsteve1** [comment]
Bumping for visibility and @tomaka once again

File: src/lib.rs [hunk 2]
```diff
@@ -369,18 +415,11 @@ impl<F> Server<F> where F: Send + Sync + 'static + Fn(&Request) -> Response {
     /// # Example
     ///
     /// ```no_run
```
Comment (tomaka) [inline src/lib.rs]:
```suggestion
    /// ```no_run
    /// use rouille::Server;
    /// use rouille::Response;
    ///
    /// let server = Server::new("localhost:0", |request| {
    ///     Response::text("hello world")
    /// }).unwrap();
```

**Comment by tomaka** [comment]
Well, I pushed a commit to your branch that updates it over master. But I screwed up by breaking the example, and now I don't have the authorization to push to your branch anymore. Sorry about that.

**Comment by rushsteve1** [comment]
I updated it to fix the test, but now some random macro breaks on the old version of Rust you have it testing against so the CI fails.

**Pull request merged**